src/chainstate/burn/operations/stack_stx.rs

```diff
@@ -196,12 +199,28 @@ impl StackStxOp {
             return None;
         }
 
+        if data.len() > 77 {
+            // too long
+            warn!(
+                "StacksStxOp payload is too long ({} bytes, expected <= {}); truncating to correct length",
+                data.len(),
+                77
+            );
+            data.truncate(77);
```

Comment:
Same idea here -- I don't think you should modify `data` if you can just get away with taking a slice.